elements/rh-video/rh-video.ts

```diff
@@ -0,0 +1,258 @@
+ * @slot consent-button-text - text for CTA button to update preferences, e.g. "Update preferences"
+ * @slot caption - optional caption below video
+ * @slot autoplay - DO NOT USE! (Used by `rh-video`.)
+ * @csspart figure - The outer container for rh-video
+ * @csspart video - The container for the video, thumbnail, and play button
+ * @csspart consent-body - The container for the consent message and consent button
+ * @csspart play - The play button on top of the thumbnail
+ * @csspart caption - The container for the caption
+ */
+@customElement('rh-video')
+export class RhVideo extends LitElement {
+  static readonly styles = [styles];
+
+  static readonly shadowRootOptions: ShadowRootInit = {
+    ...LitElement.shadowRootOptions,
+    delegatesFocus: true,
+  };
+
+  /**
+   * Whether video requires consent consent for cookies
+   */
+  @property({ type: Boolean, attribute: 'require-consent' }) requireConsent = false;
+
+  /**
+   * Whether consent consent for cookies
+   */
+  @property({ type: Boolean }) hasConsent = false;
+
+  /**
+   * Sets color theme based on parent context
+   */
+  @colorContextConsumer() private on?: ColorTheme;
+
+  // TODO(bennyp): https://lit.dev/docs/data/context/#content
+  @state() private _consentClicked = false;
+  @state() private _playClicked = false;
+  @state() private _playStarted = false;
+
+  #slots = new SlotController(this, 'caption', 'thumbnail', null);
+  #iframe: HTMLIFrameElement | undefined;
+
+  get #showConsent() {
+    return this.requireConsent && !this.hasConsent;
+  }
+
+  get consentButton() {
+    return this.shadowRoot?.querySelector('#consent-button');
+  }
+
+  get consentClicked() {
+    return this._consentClicked;
+  }
+
+  get focusableElement() {
+    let el: HTMLElement | undefined;
+    if (this.#showConsent ) {
+      el = this.consentButton as HTMLElement;
+    } else if (this.playClicked) {
+      el = this.iframeElement as HTMLElement;
+    } else {
+      el = this.playButton as HTMLElement;
+    }
+    return el;
+  }
+
+  get iframeElement() {
+    return this.#iframe;
+  }
+
+  get playButton() {
+    return this.shadowRoot?.querySelector('#play');
+  }
+
+  get playClicked() {
+    return this._playClicked;
+  }
+
+  get playStarted() {
+    return this._playStarted;
+  }
+
+  render() {
+    const { playClicked, on = '' } = this;
+    const dark = on === 'dark';
+    const svgFill = dark ? '#151515' : 'white';
+    const svgOpacity = dark ? '0.5' : '0.25';
+    const hasCaption = this.#slots.hasSlotted('caption');
+    const hasThumbnail = this.#slots.hasSlotted('thumbnail');
+    const playLabel = this.iframeElement && this.iframeElement.title ? `${this.iframeElement.title} (play video)` : 'Play video';
+    const show = this.#showConsent ? 'consent' : !!playClicked || !hasThumbnail ?
+      'video' : 'thumbnail';
+
+    return html`
+      <figure part="figure" class="${classMap({ [show]: !!show, [on]: !!on })}">
```

Comment:
@adamjohnson checking that this won't be too chatty

Reply:
Tested in VO. Seems good to me. 👍